### PR TITLE
Bug 1203455 - Update keyboard height and seamlessly calculate heights w/o reflow

### PR DESCRIPTION
**Pull request by timdream**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1b0136554e8a9271e9f270096910cf25e6054bae

File: apps/keyboard/style/keyboard.css [hunk 1]
```diff
@@ -13,31 +19,18 @@ button::-moz-focus-inner {
...
-  display: none;
-}
-
 /* Keyboard */
 #keyboard {
   position: absolute;
   bottom: 0;
   z-index: 15;
   width: 100%;
-  border-top: 0.1rem solid #a1a5a8;
```
Comment (timdream) [inline apps/keyboard/style/keyboard.css]: Border removed.

File: apps/keyboard/style/keyboard.css [hunk 2]
```diff
@@ -541,8 +548,7 @@ bubble above the key when you tap and hold. */
   height: 3rem;
   width: 100%;
   background: #333;
-  margin: -0.6rem 0 0 0; /* I don't know why we need the top margin */
```
Comment (timdream) [inline apps/keyboard/style/keyboard.css]: ... which is covered by the negative margin.

File: apps/keyboard/style/keyboard.css [hunk 3]
```diff
@@ -1,4 +1,10 @@
 html {
+  /**
+   * font-size is to be overwritten by JavaScript here.
+   * We are absuing CSS unit "rem" as a grid to describe the size of our elements.
```
Comment (janjongboom) [inline apps/keyboard/style/keyboard.css]: typo: absuing / abusing.

I don't think it's abuse to use rem like this, but soit.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=77536e2f4df22f9423062b0fe96654f03dc4aaca

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f7b389fda35e0aac1027b00aeb05fc882face23c

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7cbc9eb962ff2f55138f347fa3787796a33469ea

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7f5c8c24907a0e3b46362924cc69f78184f85493

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8695d90e11b3919d3758fb67b5612f4c62dbb259

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c37224a144c1575e3bdb07fb998b47eec2dafef6

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=11947e4ced5c8bc9d03bc3e56dc47e61a12cef7d

**Pull request merged**